### PR TITLE
cli/sql: new command-line flag `--embedded`

### LINKED ISSUE #52059: Add flag to hide welcome text in interactive sql shell
We are starting to experiment with in-browser interactive tutorials. In cases where we want to immediately load up a sql shell for the tutorial, it would be nice to have a flag that lets you hide all of the welcome text and just show the prompt. In such cases, the welcome text is distracting and could even lead users to try things that aren't the focus of the tutorial. 

In addition to a flag for hiding welcome text, perhaps it'd be nice to have a shell option to reveal that information. This would even be helpful in cases where the welcome text is displayed.

### PR DESCRIPTION
**Pull request by knz**
First commit from #59749 
Fixes #52059. 

Requested by @jseldess 

`cockroach sql --embedded`:

    #
    # Welcome to the CockroachDB SQL shell.
    # All statements must be terminated by a semicolon.
    # To exit, type: \q.
    #
    #
    # Enter \? for a brief introduction.
    #
    root@:26257/defaultdb>

`cockroach demo --embedded`:

    #
    # Welcome to the CockroachDB demo database!
    #
    # You are connected to a temporary, in-memory CockroachDB cluster of 1 node.
    #
    # Beginning initialization of the movr dataset, please wait...
    #
    # The cluster has been preloaded with the "movr" dataset
    # (MovR is a fictional vehicle sharing company).
    #
    # Reminder: your changes to data stored in the demo session will not be saved!
    #
    # The user "demo" with password "demo16754" has been created. Use it to access the Web UI!
    #
    # Enter \? for a brief introduction.
    #
    demo@127.0.0.1:26257/movr>

Release note (cli change): The SQL shell (`cockroach sql` / `demo`)
now supports a flag `--embedded`, for use with playground-style
web applications. Help text for this flag:

> Simplify/reduce the SQL CLI output to make it appropriate for
> embedding in a 'playground'-type environment.
>
> This causes the shell to omit informational message about
> aspects that can only be changed with command-line flags
> or environment variables: in an embedded environment, the user
> has no control over these and the messages would thus be
> confusing.
>
> It also causes the shell to omit informational messages about
> networking details (e.g. server address), as it is assumed
> that the embedding environment will report those instead.

### CODE REVIEW

File: pkg/cli/cliflags/flags.go [hunk 1]
```diff
@@ -291,6 +291,23 @@ transaction state. Equivalent to --echo-sql, \unset check_syntax and
 \set prompt1 %n@%M>.`,
 	}
 
+	EmbeddedMode = FlagInfo{
+		Name: "embedded",
+		Description: `
+Simplify/reduce the SQL CLI output to make it appropriate for
```
Comment (otan) [inline pkg/cli/cliflags/flags.go:297]: nit: pick simplify or reduce

File: pkg/cli/demo.go [hunk 2]
```diff
@@ -229,11 +229,15 @@ func checkDemoConfiguration(
...
+			demoCtx.nodes = 9
+			if !sqlCtx.embeddedMode {
+				// Only explain how the configuration was interpreted if the
+				// user has control over it.
+				const msg = `#
 # --geo-partitioned replicas operates on a 9 node cluster.
 # The cluster size has been changed from the default to 9 nodes.`
-			fmt.Println(msg)
-			demoCtx.nodes = 9
+				fmt.Println(msg)
```
Comment (otan) [inline pkg/cli/demo.go:239]: i do wonder if we should make a new function `PrintlnIfNotEmbedded` or something to simplify the if checks. but optional.

**Review by otan — APPROVED** [review]

File: pkg/cli/demo.go (hunk 2 above)
Comment (knz) [reply pkg/cli/demo.go:239]: I like the idea! I'll try it out.

**Review by knz — COMMENTED** [review]
Thanks Oliver for the review. Also awaiting for feedback from @jseldess to confirm this answers the use case.

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/59750#-:-MSbukAX2vPql16p5BFB:bf51ynf)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @jseldess and @otan)

___
*[pkg/cli/demo.go, line 239 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/59750#-MSbTB3ne1nsb6be17xt-r2-239:-MSbuQeZ9CKsOVjDGreF:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/00f79af14d1db33fdc81ffc0179205079bb884c3/pkg/cli/demo.go#L239)):*
<details><summary><i>Previously, knz (kena) wrote…</i></summary><blockquote>

I like the idea! I'll try it out.
</blockquote></details>

Done.
___
*[pkg/cli/cliflags/flags.go, line 297 at r2](https://reviewable.io/reviews/cockroachdb/cockroach/59750#-MSbTB3ne1nsb6be17xs-r2-297:-MSbtnnD7Cm6d2QV9luI:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/00f79af14d1db33fdc81ffc0179205079bb884c3/pkg/cli/cliflags/flags.go#L297)):*
<details><summary><i>Previously, otan (Oliver Tan) wrote…</i></summary><blockquote>

nit: pick simplify or reduce
</blockquote></details>

Done.

**Review by jseldess — APPROVED** [review]
LGTM!

Just tested and it works like a charm:

```
~/go/src/github.com/cockroachdb/cockroach$ ./cockroachshort demo --embedded
#
# Welcome to the CockroachDB demo database!
#
# You are connected to a temporary, in-memory CockroachDB cluster of 1 node.
#
# Beginning initialization of the movr dataset, please wait...
#
# The cluster has been preloaded with the "movr" dataset
# (MovR is a fictional vehicle sharing company).
#
# Reminder: your changes to data stored in the demo session will not be saved!
#
# The user "demo" with password "demo65926" has been created. Use it to access the Web UI!
#
#
# Enter \? for a brief introduction.
#
demo@127.0.0.1:26257/movr>
```

**Comment by knz** [comment]
Thanks to both!

bors r=otan,jseldess

**Pull request merged**